UploadDaemon_Test/MockUpload.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Threading.Tasks;
+using UploadDaemon.Upload;
+using UploadDaemon.SymbolAnalysis;
+
+/// <summary>
+/// Mock for the IUpload interface for testing.
+/// </summary>
+public class MockUpload : IUpload
```

Comment:
For the purpose of the merger tests this could be be replaced by the less verbose `Mock` instance.

```csharp
var upload1 = new Mock<IUpload>();
upload1.Setup(u => u.UploadAsync(any, any)).Return(Task.FromResult(true)); // * 
upload1.Setup(u => u. UploadLineCoverageAsync(any, any, any)).Return(Task.FromResult(true)); // *
upload1.Setup(u => u.GetDictionaryKey()).Return("key1");
```

In fact, the same would work for the `UploadTaskTests`, where the `LastUsedVersion` could be verified using

```csharp
upload1.Verify(u => u.UploadAsync(any, "expected version")); // *
```

*Not sure about the arg matcher syntax from the top of my head 😇

Reply:
While the mock is less verbose in the sense that it uses less lines, I find it harder to read.
Let's best discuss this in person